src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -886,35 +884,35 @@ def call(
         # embed positions
         positions = self.embed_positions(input_shape, past_key_values_length)
 
-        if inputs_embeds is None:
-            inputs_embeds = self.embed_tokens(inputs["input_ids"])
-        else:
-            inputs_embeds = inputs["inputs_embeds"]
+        if inputs["inputs_embeds"] is None:
+            inputs["inputs_embeds"] = self.embed_tokens(inputs["input_ids"])
 
-        hidden_states = inputs_embeds * self.embed_scale
+        hidden_states = inputs["inputs_embeds"] * self.embed_scale
 
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask = None
+        combined_attention_mask = tf.fill(dims=input_shape, value=-1.0)
         if input_shape[-1] > 1:
             combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
 
+        is_combined_None = tf.reduce_any(combined_attention_mask == -1)
```

Comment:
It looks like the condition is `input_shape[-1] > 1` since `_make_causal_mask` is `never None`. I think ti should be faster than testing if the elements are all -1 since we could store it before the definition of `combined_attention_mask`.